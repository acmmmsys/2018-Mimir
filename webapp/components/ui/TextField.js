import React, { Component } from 'react';
import classnames from 'classnames';
import { compose, withState, withHandlers } from 'recompose';

import 'style/ui/TextField.scss';

const defaultClassNames = [
    'text-field-container',
];

class TextField extends Component {

    static defaultProps = {
        classNames: [],
        className: '',
    }

    shouldComponentUpdate(nextProps, nextState) {
        return this.props.value !== nextProps.value;
    }

    render() {

        let classNames = classnames(
            defaultClassNames,
            this.props.className,
        );

        return (
            <div className={classNames}>
                <label>
                    {this.props.label}
                </label>
                <input onChange={this.props.onChange}
                    type='text' 
                    value={this.props.value}
                />
            </div>
        );
    }
}


export default TextField;