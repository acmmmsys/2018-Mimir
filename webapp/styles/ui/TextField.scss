

.text-field-container {
    width: 100%;
    padding: 1px 0;

    label {
        display: inline-block;
        padding-right: 10px;
    }

    input[type="text"] {
        border: none;
        border-bottom: solid 2px white;

        &:focus {
            outline: none;
            border: none;
            border-bottom: solid 2px #c9c9c9;
        }
    }
}