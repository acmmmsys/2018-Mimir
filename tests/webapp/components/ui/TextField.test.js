import React from 'react';
import TextField from '../../../../webapp/components/ui/TextField'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('TextField does render', () => {
  const wrapper = shallow(
      <TextField store={mockStore(initialState)}>This is a test!</TextField>
  );
  expect(wrapper).toMatchSnapshot();
});
