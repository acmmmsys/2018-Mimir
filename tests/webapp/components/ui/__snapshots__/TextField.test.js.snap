// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField does render 1`] = `
<div
  className="text-field-container"
>
  <label />
  <input
    type="text"
  />
</div>
`;
